RIOT-OS/RIOT
# cpu/esp32: reimplementation of module esp_can as module periph can

### Contribution description

This PR is a refactoring of the ESP32 CAN controller driver implementation. Refactoring became necessary to follow the approach as introduced with PR #6178, which defines CAN interfaces as peripherals with automatic configuration.

Accordingly, the ESP32 CAN controller driver is no longer a separate module (`esp_can`), but rather part of the peripheral drivers which is enabled when module `periph_can` is used.

### Testing procedure

For testing, an ESP32 board with a connected CAN transceiver module is required. The only supported board is which already has such a CAN transceiver module on board, is the [Olimex ESP32-EVB](http://doc.riot-os.org/group__boards__esp32__olimex-esp32-evb.html). Therefore, it is suggested to use this board.

After flashing the board with application `tests/conn_can`
```
make BOARD=esp32-olimex-evb -C tests/conn_can flash term
```
the `CAN-L` and `CAN-H` have to be connected to a can bus. As counterpart a Linux box should be used with the `can-utils`.
```
linux> sudo ip link set can0 type can bitrate 500000
linux> sudo ip link set can0 up
```
After connecting to the ESP32 board in a terminal, the following tests can be executed.

1. List CAN interfaces:
   ```
   esp32> can list
   CAN #0: esp-can
   ```
2. Send a CAN frame from ESP32 to the Linux box (start `candump` first):
   ```
   linux$ candump can0
    
   can0  123   [4]  11 22 33 44
   ```
   ```
   esp32> can send 0 123 11 22 33 44
   ```
3. Send a CAN frame from Linux box to ESP32 (start `can dump` first and send within 10 seconds):
   ```
   esp32> can dump 0 0 10000
   esp-can (0)      5A1  [8]  11 22 33 44 55 66 77 88
   ```
   ```
   linux$ cansend can0 5A1#11.2233.44556677.88
   ```
### Issues/PRs references

Implements #6178 for ESP32.
